pennylane_lightning/core/src/simulators/lightning_tensor/cutn/CutnBase.hpp

```diff
@@ -0,0 +1,222 @@
+        initHelper_();
+    }
+
+    CutnBase(const size_t numQubits, int device_id = 0,
+             cudaStream_t stream_id = 0)
+        : BaseType(numQubits), handle_(make_shared_cutn_handle()),
+          dev_tag_({device_id, stream_id}) {
+        initHelper_();
+    }
+
+    virtual ~CutnBase() {
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetDestroyState(quantumState_));
+    }
+
+    /**
+     * @brief Get the cutensornet handle that the object is using.
+     *
+     * @return cutensornetHandle_t
+     */
+    [[nodiscard]] auto getCutnHandle() const -> cutensornetHandle_t {
+        return handle_.get();
+    }
+
+    /**
+     * @brief Get the quantum state pointer.
+     *
+     * @return cutensornetState_t
+     */
+    [[nodiscard]] auto getQuantumState() -> cutensornetState_t {
+        return quantumState_;
+    };
+
+    /**
+     * @brief Get device and Cuda stream information (device ID and the
+     * associated Cuda stream ID).
+     *
+     * @return DevTag
+     */
+    [[nodiscard]] auto getDevTag() -> DevTag<int> & { return dev_tag_; }
+
+    /**
+     * @brief Get the memory size used for a work space
+     *
+     * @return size_t
+     */
+    size_t getWorkSpaceMemorySize(cutensornetWorkspaceDescriptor_t &workDesc) {
+        int64_t worksize{0};
+
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetWorkspaceGetMemorySize(
+            /* const cutensornetHandle_t */ getCutnHandle(),
+            /* cutensornetWorkspaceDescriptor_t */ workDesc,
+            /* cutensornetWorksizePref_t */
+            CUTENSORNET_WORKSIZE_PREF_RECOMMENDED,
+            /* cutensornetMemspace_t*/ CUTENSORNET_MEMSPACE_DEVICE,
+            /* cutensornetWorkspaceKind_t */ CUTENSORNET_WORKSPACE_SCRATCH,
+            /*  int64_t * */ &worksize));
+
+        return worksize;
+    }
+
+    /**
+     * @brief Set the memory for a work space
+     *
+     * @param workDesc cutensornet work space descriptor
+     * @param scratchPtr Pointer to scratch memory
+     * @param worksize Memory size of a work space
+     */
+    void setWorkSpaceMemory(cutensornetWorkspaceDescriptor_t &workDesc,
+                            void *scratchPtr, int64_t &worksize) {
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetWorkspaceSetMemory(
+            /* const cutensornetHandle_t */ getCutnHandle(),
+            /* cutensornetWorkspaceDescriptor_t */ workDesc,
+            /* cutensornetMemspace_t*/ CUTENSORNET_MEMSPACE_DEVICE,
+            /* cutensornetWorkspaceKind_t */ CUTENSORNET_WORKSPACE_SCRATCH,
+            /* void *const */ scratchPtr,
+            /* int64_t */ worksize));
+    }
+
+    /**
+     * @brief Save quantumState information to data provided by a user
+     *
+     * @param extentsPtr Extents of each sites
+     * @param tensorPtr Pointer to tensors provided by a user
+     */
+    void computeState(std::vector<int64_t *> &extentsPtr,
+                      std::vector<void *> &tensorPtr) {
+        cutensornetWorkspaceDescriptor_t workDesc;
```

Comment:
Do you think this should be method-level variable, or should it be a private class property that the methods can reference?

Reply:
Since this could be only called by exact and MPS backends, let's as it as `protected`.